src/main/java/seedu/address/storage/Storage.java

```diff
@@ -27,31 +25,102 @@
     void saveUserPrefs(ReadOnlyUserPrefs userPrefs) throws IOException;
 
     /**
-     * Gets the address book storage.
+     * Returns the file path of the application data file.
+     */
+    Path getApplicationItemListFilePath();
+
+    /**
+     * Returns application ItemList data as a {@link ItemList}.
+     * Returns {@code Optional.empty()} if storage file is not found.
      *
-     * @return address book storage.
+     * @throws DataConversionException if the data in storage is not in the expected format.
+     * @throws IOException             if there was any problem when reading from the storage.
      */
-    ListStorage<Person, JsonAdaptedPerson> getAddressBookStorage();
+    Optional<ReadOnlyItemList<ApplicationItem>> readApplicationItemList() throws DataConversionException, IOException;
 
     /**
-     * Gets the application item list storage.
+     * @see #getApplicationItemListFilePath()
```

Comment:
I dont know if you need a fullstop here?

Reply:
Hmm, it's copy and pasted from AB3's. Hahaha.